envoy/api/v2/discovery.proto

```diff
@@ -62,6 +66,14 @@ message DiscoveryResponse {
   // The response resources. These resources are typed and depend on the API being called.
   repeated google.protobuf.Any resources = 2 [(gogoproto.nullable) = false];
 
+  // When true the resources list is a diff to the exiting resources.
+  // Incremental can only be set to true in DiscoveryResponse if this stream has
+  // seen incremental = true in a DiscorevyRequest.
```

Comment:
I don't see why there needs to be a connection here. Logically, the resource names are a hint only, and we could do incremental resource names with or without incremental updates from the server.

We need to also discuss how incremental and full updates can be combined (e.g. sometimes it might make sense to do a full update rather than tons of tiny incremental ones or when establishing known state initially). The xDS protocol RFC is essentially https://github.com/envoyproxy/data-plane-api/blob/master/XDS_PROTOCOL.md, so we can cover some of this there.

Also nit: s/DiscorevyRequest/DiscoveryRequest/.

Reply:
+1 on incremental vs. full. I suppose it's possible w/ this API, but per my above comment should we consider a more specific sequence of events in which on initial connection we expect the server to send the full dump, then switch to incremental mode? Either way, given this flag, Envoy can either do the delta diff or not.